tiledb/api/src/array/mod.rs

```diff
@@ -1019,4 +1130,102 @@ pub mod tests {
         test_uri.close().map_err(|e| Error::Other(e.to_string()))
     }
+
+    fn create_simple_dense(
+        test_uri: &dyn TestArrayUri,
+        ctx: &Context,
+    ) -> TileDBResult<String> {
+        let domain = {
+            let rows = DimensionBuilder::new(
+                ctx,
+                "id",
+                Datatype::Int32,
+                ([1, 410], 10),
+            )?
+            .build();
+
+            DomainBuilder::new(ctx)?.add_dimension(rows)?.build()
+        };
+
+        let schema = SchemaBuilder::new(ctx, ArrayType::Dense, domain)?
+            .add_attribute(
+                AttributeBuilder::new(ctx, "a", Datatype::Int32)?.build(),
+            )?
+            .build()?;
+
+        let array_uri = test_uri
+            .with_path("quickstart_dense")
+            .map_err(|e| Error::Other(e.to_string()))?;
+        Array::create(ctx, &array_uri, schema)?;
+        Ok(array_uri)
+    }
+
+    fn write_dense_vector_4_fragments(
+        ctx: &Context,
+        array_uri: &str,
+        timestamp: u64,
+    ) -> TileDBResult<()> {
+        // Subarray boundaries.
+        let boundaries = [0, 200, 250, 310, 410];
+
+        for i in 0..4 {
+            // Prepare cell buffer.
+            let low_bound = boundaries[i];
+            let high_bound = boundaries[i + 1];
+
+            let mut data: Vec<i32> = vec![];
+            for j in low_bound..high_bound {
+                data.push(j);
+            }
+
+            let opener = ArrayOpener::new(ctx, array_uri, Mode::Write)?
+                .end_timestamp(timestamp + i as u64 + 1)?;
+            let array = opener.open()?;
+
+            let q1 = WriteBuilder::new(array)?
+                .layout(QueryLayout::RowMajor)?
+                .start_subarray()?
+                .set_subarray(&[low_bound + 1, boundaries[i + 1]])?
+                .finish_subarray()?
+                .data_typed("a", &data)?
+                .build();
+            q1.submit().and_then(|_| q1.finalize())?;
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_array_consolidation() -> TileDBResult<()> {
+        // Test advanced consolidation. Based on unit-capi-consolidation.cc.
+
+        let ctx: Context = Context::new().unwrap();
+        let array_uri = tiledb_test_utils::get_uri_generator().unwrap();
+        let array_uri = create_simple_dense(&array_uri, &ctx)?;
+        write_dense_vector_4_fragments(&ctx, &array_uri, 0).unwrap();
+
+        let mut config = Config::new()?;
+        config.set("sm.consolidation.steps", "1").unwrap();
+        config.set("sm.consolidation.step_min_frags", "2").unwrap();
+        config.set("sm.consolidation.step_max_frags", "2").unwrap();
+        config
+            .set("sm.consolidation.step_size_ratio", "0.0")
+            .unwrap();
+        config.set("sm.consolidation.buffer_size", "10000").unwrap();
+
+        let count_fragment_fn = || -> TileDBResult<u32> {
+            FragmentInfoBuilder::new(&ctx, array_uri.clone())?
```

Comment:
This is a little odd. I know it's the way to do it with the current API but I feel like we should have `FragmentInfoList::open` or something.

Reply:
Oh, I agree that this is the best that can be done with the current way of the world, I just thing we should improve the way of the world at some point.